app/services/create_target_service.rb

```diff
@@ -0,0 +1,48 @@
+class CreateTargetService
+  def initialize(current_user, params)
+    @current_user = current_user
+    @params = params
+  end
+
+  def create!
+    Target.transaction do
+      target = targets.create!(@params)
+      check_matchs(target)
+      target
+    end
+  end
+
+  private
+
+  def targets
+    @targets ||= @current_user.targets
+  end
+
+  def check_matchs(new_target)
+    matched_target = find_target_to_match(new_target)
+    return if matched_target.nil?
+
+    matched_target.update!(matched: true)
+    new_target.update!(matched: true)
```

Comment:
Just asking here, what does it mean that a target is `matched`? Is it able to match again?